include/ensmallen_bits/problems/fonseca_fleming_function.hpp

```diff
@@ -71,19 +71,23 @@ class FonsecaFlemingFunction
   {
     typename MatType::elem_type Evaluate(const MatType& coords)
     {
-        return 1.0f - exp(-pow(coords[0] - 1.0f / sqrt(3), 2) -
-            - pow(coords[1] - 1.0f / sqrt(3), 2)
-            - pow(coords[2] - 1.0f / sqrt(3), 2));
+        return 1.0 - exp(
+             -pow(static_cast<double>(coords[0]) - 1.0 / sqrt(3.0), 2.0)
```

Comment:
Looking at the error message: `‘sqrt(int)’ is ambiguous` I think the `static_cast` is not necessary?

Reply:
It's probably overkill, I can remove if need be.

Reply:
We don't actually know what `coords[0]` is, so I might suggest using `MatType::elem_type` maybe like this:

```
typedef MatType::elem_type T;
return T(1) - exp(-pow(coords[0] - T(1) / sqrt(T(3)), T(2))
    - pow(coords[1] - T(1) / sqrt(T(3)), T(2))
    - pow(coords[2] - T(1) / sqrt(T(3)), T(2)));
```

Or, something like that could work.  Basically the idea is to force every internal type inside the expression to be the same as `MatType::elem_type`.

Reply:
@rcurtin I agree with a type cast. That said, we input into `pow()` and `sqrt()` may be either `float`, `double`, or `long double` per their function definitions.

So, `MatType::elem_type` includes `unsigned int` and `int`. 

short | | long 
-- | -- | --
mat |= | `Mat<double>`
dmat | = | `Mat<double>`
fmat | = | `Mat<float>`
cx_mat | = | `Mat<cx_double>`
cx_dmat | = | `Mat<cx_double>`
cx_fmat | = | `Mat<cx_float>`
umat | = | `Mat<uword>`
imat | = | `Mat<sword>`

Reply:
Ah, good point!  In this case, I'd be fine forcing to `double` like you suggested.